pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotApplicationQuotaRestletResource.java

```diff
@@ -0,0 +1,163 @@
+
+import static org.apache.pinot.spi.utils.CommonConstants.SWAGGER_AUTHORIZATION_KEY;
+
+
+@Api(tags = Constants.DATABASE_TAG, authorizations = {@Authorization(value = SWAGGER_AUTHORIZATION_KEY)})
+@SwaggerDefinition(securityDefinition = @SecurityDefinition(apiKeyAuthDefinitions = {
+    @ApiKeyAuthDefinition(name = HttpHeaders.AUTHORIZATION, in = ApiKeyAuthDefinition.ApiKeyLocation.HEADER, key =
+        SWAGGER_AUTHORIZATION_KEY, description =
+        "The format of the key is  ```\"Basic <token>\" or \"Bearer "
+            + "<token>\"```"), @ApiKeyAuthDefinition(name = CommonConstants.APPLICATION, in =
+    ApiKeyAuthDefinition.ApiKeyLocation.HEADER, key = CommonConstants.APPLICATION, description =
+    "Application context passed through http header. If no context is provided 'default' application "
+        + "context will be considered.")
+}))
+@Path("/")
+public class PinotApplicationQuotaRestletResource {
+  public static final Logger LOGGER = LoggerFactory.getLogger(PinotApplicationQuotaRestletResource.class);
+
+  @Inject
+  PinotHelixResourceManager _pinotHelixResourceManager;
+
+  /**
+   * API to get application quota configs. Will return null if application quotas are not defined
+   */
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/applicationQuotas")
+  @Authorize(targetType = TargetType.CLUSTER, action = Actions.Cluster.GET_APPLICATION_QUERY_QUOTA)
+  @ApiOperation(value = "Get all application qps quotas", notes = "Get all application qps quotas")
+  public Map<String, Double> getApplicationQuotas(@Context HttpHeaders httpHeaders) {
+    Map<String, Double> quotas = _pinotHelixResourceManager.getApplicationQuotas();
+    if (quotas != null) {
+      return quotas;
+    }
+
+    HelixConfigScope scope = new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.CLUSTER).forCluster(
+        _pinotHelixResourceManager.getHelixClusterName()).build();
+    HelixAdmin helixAdmin = _pinotHelixResourceManager.getHelixAdmin();
+    String defaultQuota =
+        helixAdmin.getConfig(scope, Collections.singletonList(CommonConstants.Helix.APPLICATION_MAX_QUERIES_PER_SECOND))
+            .getOrDefault(CommonConstants.Helix.APPLICATION_MAX_QUERIES_PER_SECOND, null);
+
+    quotas = new HashMap<>();
+    quotas.put(CommonConstants.DEFAULT_APPLICATION, defaultQuota != null ? Double.parseDouble(defaultQuota) : null);
+    return quotas;
+  }
+
+  /**
+   * API to get application quota configs. Will return null if application quotas are not defined
+   */
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/applicationQuotas/{appName}")
+  @Authorize(targetType = TargetType.CLUSTER, action = Actions.Cluster.GET_APPLICATION_QUERY_QUOTA)
+  @ApiOperation(value = "Get application qps quota", notes = "Get application qps quota")
+  public Double getApplicationQuota(@Context HttpHeaders httpHeaders, @PathParam("appName") String appName) {
+    if (!appName.equals(extractApplicationFromHttpHeaders(httpHeaders))) {
```

Comment:
Is there a specific usecase to allow application context through headers? If so we also need to look for the application header in the query path as well

Reply:
I don't know. This check was taken from db quota restlet resource.

Reply:
Oh in that case let's refrain from passing the application context as header. Agree that we are allowing to pass database as header but later there was a concern raised suggesting to restrict the configs passed while querying to be restricted to query options.

Reply:
ok, removed.